Client/WebAuthN/U2FExtensions.swift

```diff
@@ -121,83 +130,145 @@ class U2FExtensions: NSObject {
         didSet {
             if oldValue == observeSessionStateUpdates {
                 return
             }
             
-            let keySession = YubiKitManager.shared.keySession as AnyObject
+            guard let accessorySession = YubiKitManager.shared.accessorySession as? YKFAccessorySession else {
+                return
+            }
             
             if observeSessionStateUpdates {
-                keySession.addObserver(self, forKeyPath: #keyPath(YKFKeySession.sessionState), options: [.new, .old], context: &U2FExtensions.observationContext)
+                accessorySessionStateObservation = accessorySession.observe(\.sessionState, changeHandler: { [weak self] session, change in
+                    ensureMainThread {
+                        self?.observeSessionStateUpdates = false
+                        if session.sessionState != .closed {
+                            self?.keyType = .accessory
+                        }
+                        self?.handleSessionStateChange()
+                    }
+                })
             } else {
-                keySession.removeObserver(self, forKeyPath: #keyPath(YKFKeySession.sessionState))
+                accessorySessionStateObservation = nil
             }
         }
     }
     
-    private var observeKeyStateUpdates: Bool = false {
+    private var observeServiceStateUpdates: Bool = false {
         didSet {
-            if oldValue == observeKeyStateUpdates {
+            if oldValue == observeServiceStateUpdates {
                 return
             }
             
-            let keySession = YubiKitManager.shared.keySession as AnyObject
+            let keySession = YubiKitManager.shared.accessorySession as AnyObject
             
-            if observeKeyStateUpdates {
-                keySession.addObserver(self, forKeyPath: #keyPath(YKFKeySession.fido2Service.keyState), options: [], context: &U2FExtensions.observationContext)
-                keySession.addObserver(self, forKeyPath: #keyPath(YKFKeySession.u2fService.keyState), options: [], context: &U2FExtensions.observationContext)
+            if observeServiceStateUpdates {
+                keySession.addObserver(self, forKeyPath: #keyPath(YKFAccessorySession.fido2Service.keyState), options: [], context: &U2FExtensions.observationContext)
+                keySession.addObserver(self, forKeyPath: #keyPath(YKFAccessorySession.u2fService.keyState), options: [], context: &U2FExtensions.observationContext)
+            } else {
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.fido2Service.keyState))
+                keySession.removeObserver(self, forKeyPath: #keyPath(YKFAccessorySession.u2fService.keyState))
+            }
+        }
+    }
+    
+    private var keyType: FIDO2ViewControllerKeyType = .unknown
+    
+    /// Returns the service associated with the desired session.
+    private var keyFido2Service: YKFKeyFIDO2ServiceProtocol? {
+        get {
```

Comment:
@iccub - I used the closure to return the `fido2Service` depending on the `keyType`. I can convert this to a function but a get closure felt cleaner. Let me know what you think.

Reply:
I know, you can literally just remove the `get` keyword, and leave the rest

so it's
```
private var keyFido2Service: YKFKeyFIDO2ServiceProtocol? {
    // code logic
}
```

instead of 
```
private var keyFido2Service: YKFKeyFIDO2ServiceProtocol? {
    get {
        // code logic
    }
}
```

per Swift documentation
```
You can simplify the declaration of a read-only computed property by removing the get keyword and its braces:
```